### PR TITLE
UI: force gtk tabs to be justified for preferences and equalizers

### PR DESCRIPTION
**Pull request by aurelienpierre**

### CODE REVIEW

File: tools/generate_prefs.xsl [hunk 1]
```diff
@@ -331,14 +332,14 @@
...
 
-	<xsl:template match="dtconfig[type='dir']" mode="reset">
-		<xsl:text>
-			dt_conf_set_string("</xsl:text><xsl:value-of select="name"/><xsl:text>", "</xsl:text><xsl:value-of select="default"/><xsl:text>");
-			gchar *folder = dt_conf_get_string("</xsl:text><xsl:value-of select="name"/><xsl:text>");
-			gtk_file_chooser_set_filename(GTK_FILE_CHOOSER(widget), folder);
-			g_free(folder);
-		</xsl:text>
-	</xsl:template>
+  <xsl:template match="dtconfig[type='dir']" mode="reset">
```
Comment (aurelienpierre) [inline tools/generate_prefs.xsl]: tabs indents converted to spaces - sorry, Geany is pig-headed about that

File: tools/generate_prefs.xsl [hunk 2]
```diff
@@ -376,13 +377,13 @@
...
   </xsl:template>
 
-	<xsl:template match="dtconfig[type='dir']" mode="change">
-		<xsl:text>
-		gchar *folder =	gtk_file_chooser_get_filename(GTK_FILE_CHOOSER(widget));
-		dt_conf_set_string("</xsl:text><xsl:value-of select="name"/><xsl:text>", folder);
-		g_free(folder);
-		</xsl:text>
-	</xsl:template>
+  <xsl:template match="dtconfig[type='dir']" mode="change">
```
Comment (aurelienpierre) [inline tools/generate_prefs.xsl]: tabs indents converted to spaces - sorry, Geany is pig-headed about that

File: tools/generate_prefs.xsl [hunk 3]
```diff
@@ -400,10 +401,10 @@
   </xsl:template>
 
   <xsl:template match="dtconfig[type='dir']" mode="tab">
-		<xsl:text>    widget = gtk_file_chooser_button_new(_("select directory"), GTK_FILE_CHOOSER_ACTION_SELECT_FOLDER);
-		gtk_file_chooser_button_set_width_chars(GTK_FILE_CHOOSER_BUTTON(widget), 20);
-		gtk_widget_set_hexpand(widget, TRUE);
-		gtk_widget_set_halign(widget, GTK_ALIGN_FILL);
+    <xsl:text>    widget = gtk_file_chooser_button_new(_("select directory"), GTK_FILE_CHOOSER_ACTION_SELECT_FOLDER);
```
Comment (aurelienpierre) [inline tools/generate_prefs.xsl]: tabs indents converted to spaces - sorry, Geany is pig-headed about that

File: tools/generate_prefs.xsl [hunk 4]
```diff
@@ -68,7 +69,7 @@
 
   <xsl:for-each select="./dtconfiglist/dtconfig[@prefs]">
     <xsl:if test="name != 'opencl' or $HAVE_OPENCL=1">
-      <xsl:text>static void&#xA;preferences_response_callback_</xsl:text><xsl:value-of select="generate-id(.)"/><xsl:text> (GtkDialog *dialog, gint response_id, GtkWidget *widget)&#xA;{&#xA;  if(response_id != GTK_RESPONSE_ACCEPT)&#xA;    return;&#xA;</xsl:text>
+      <xsl:text>static void&#xA;preferences_response_callback_</xsl:text><xsl:value-of select="generate-id(.)"/><xsl:text> (GtkDialog *dialog, gint response_id, GtkWidget *widget)&#xA;{&#xA;  if(response_id != GTK_RESPONSE_ACCEPT) return;&#xA;</xsl:text>
```
Comment (aurelienpierre) [inline tools/generate_prefs.xsl]: changed because GCC complained about possibly misleading indents

File: src/iop/filmicrgb.c [hunk 5]
```diff
@@ -1435,9 +1435,7 @@ void gui_init(dt_iop_module_t *self)
   gtk_widget_show_all(GTK_WIDGET(gtk_notebook_get_nth_page(g->notebook, 0)));
   gtk_box_pack_start(GTK_BOX(self->widget), GTK_WIDGET(g->notebook), FALSE, FALSE, 0);
 
-  gtk_container_child_set(GTK_CONTAINER(g->notebook), page1, "tab-expand", TRUE, "tab-fill", TRUE, NULL);
-  gtk_container_child_set(GTK_CONTAINER(g->notebook), page2, "tab-expand", TRUE, "tab-fill", TRUE, NULL);
-  gtk_container_child_set(GTK_CONTAINER(g->notebook), page3, "tab-expand", TRUE, "tab-fill", TRUE, NULL);
+  dtgtk_justify_notebook_tabs(g->notebook);
```
Comment (aurelienpierre) [inline src/iop/filmicrgb.c]: this change is equivalent

File: src/iop/toneequal.c [hunk 6]
```diff
@@ -3126,9 +3126,7 @@ void gui_init(struct dt_iop_module_t *self)
   gtk_box_pack_start(GTK_BOX(self->widget), GTK_WIDGET(g->notebook), FALSE, FALSE, 0);
   g_signal_connect(G_OBJECT(g->notebook), "button-press-event", G_CALLBACK(notebook_button_press), self);
 
-  gtk_container_child_set(GTK_CONTAINER(g->notebook), page1, "tab-expand", TRUE, "tab-fill", TRUE, NULL);
-  gtk_container_child_set(GTK_CONTAINER(g->notebook), page2, "tab-expand", TRUE, "tab-fill", TRUE, NULL);
-  gtk_container_child_set(GTK_CONTAINER(g->notebook), page3, "tab-expand", TRUE, "tab-fill", TRUE, NULL);
+  dtgtk_justify_notebook_tabs(g->notebook);
```
Comment (aurelienpierre) [inline src/iop/toneequal.c]: this change is equivalent

File: tools/generate_prefs.xsl (hunk 3 above)
Comment (TurboGit) [reply tools/generate_prefs.xsl]: In fact that's a fix :) We don't want tabs in code, thanks.

**Comment by Nilvus** [comment]
Indeed, a good add to have the better UI possible for the 3.0 release. Much better to see tabs this way. Thanks @aurelienpierre.

**Pull request merged**